yt_dlp/extractor/common.py

```diff
@@ -3633,7 +3633,23 @@ def _set_cookie(self, domain, name, value, expire_time=None, port=None,
 
     def _get_cookies(self, url):
         """ Return a http.cookies.SimpleCookie with the cookies for the url """
-        return http.cookies.SimpleCookie(self._downloader._calc_cookies(url))
+        raw_cookie = self._downloader._calc_cookies(url)
+
+        cookie = http.cookies.SimpleCookie()
+        if raw_cookie is None:
+            return cookie
+
+        # Iterate over split chunks to skip invalid values instead of failing fast.
+        for chunk in raw_cookie.split(";"):
```

Comment:
`;` is a valid character in a `quoted-string` cookie value.
```
>>> http.cookies.SimpleCookie('''chips="a;hoy"; vienna=finger''')
<SimpleCookie: chips='a;hoy' vienna='finger'>
>>> 
```
Check https://github.com/python/cpython/blob/3.10/Lib/http/cookies.py for what is currently being done (and has been for 22 years, according to the header comments).

What could additionally be done is to add a parallel pattern to the value match in `_CookiePattern`, s/t like this
```
r"""
(?:
  (?P<val>                         # Start of group 'val'
      "(?:[^\\"]|\\.)*"                  # Any doublequoted string
      |                                  # or
      \w{3},\s[\w\d\s-]{9,11}\s[\d:]{8}\sGMT  # Special case for "expires" attr
      |                                  # or
      [""" + _LegalValueChars + r"""]*      # Any word or empty string
      )|                                # End of group 'val'
  (?P<bad>(?:\\;|[^;])*)           # fallback match to skip bad value 
)
"""
Then in a rewrite of the `BaseCookie.__parse_string()` method, add a step to check the `bad` group: if it matches, skip past the match and try to parse the cookie string from there.

Reply:
Yeah cookie parsing is fairly complicated and needs to obey the relevant RFCs, so it's not going to be simple.

Reply:
Though this does look to be what django does https://github.com/django/django/commit/93a135d111c2569d88d65a3f4ad9e6d9ad291452

Reply:
RFC2109 has been obsoleted by RFC6265 and suggests semicolons are in fact not valid as can be read in [RFC6265, Section 4.1.1](https://www.rfc-editor.org/rfc/rfc6265#section-4.1.1) about `cookie-octet` and `cookie-name` (which forwards to [RFC2616, Section 2.2, `token`](https://www.rfc-editor.org/rfc/rfc2616#section-2.2)).

Reply:
Even so, follow CPython as closely as possible, making only minimum required change to fix our issue

Reply:
In my opinion the current way of parsing the cookie string cannot easily be fixed to handle the strings as required.

This change requires subclassing `SimpleCookie` and completely reimplementing its `load()` or `__parse_string()` function with a custom regex and behavior. This new subclass will then be returned from `_get_cookies()` instead. Let me know if this is the desired behavior or, if not, how and where the implementation should be.

Reply:
I have implemented a wrapper function `__make_simple_cookie()` which closely resembles the parsing done in `http.cookies` (except for the `bad` group and not failing on errors). Please review anew.